### DISCUSSION TITLE
External sensor doesn't detect task

### DISCUSSION THREAD

**Discussion by lrahal** [apache/airflow#29720] [Q&A]
Hello everyone! 🕺🏻
I am trying to use external sensors for the first time. I want task 2 of DAG 2 to be triggered when task 1 of DAG 1 has ended. I see the external sensor poking without ever detecting the end of the task 1.
Using Airflow 2.4.3.

DAG 1:
```
with DAG(
    "pipeline",
        default_args={
        "owner": "airflow",
        "depends_on_past": False,
        "start_date": datetime.now() - timedelta(days=2),
        "schedule_interval": "0 1 * * *",
        "retries": 0,
},
    schedule_interval="0 4 * * *",
) as dag:
   end_task = BashOperator(
        task_id="end_task",
        bash_command="echo 'End of pipeline'",
        trigger_rule="all_done",
        dag=dag
    )

end_task
```

DAG 2:
```
@dag(
    default_args={
        "owner": "airflow",
        "depends_on_past": False,
        "start_date": datetime.now() - timedelta(days=2),
        "retries": 0,
    },
)
def permifrost_dag():
    run_permifrost = BashOperator(
        task_id="run_permifrost",
        bash_command="permifrost -v run /admin/permifrost.yml",
    )

    pipeline_end_sensor = ExternalTaskSensor(
        task_id="pipeline_end_task_sensor",
        external_dag_id="pipeline",
        external_task_id="end_task",
        check_existence=True,
        poke_interval=60,
        execution_delta = timedelta(minutes=3),
    )

    pipeline_end_sensor >> run_permifrost


dag = permifrost_dag()
```

Would you understand why pipeline_end_sensor never detects end_task?
Thanks a lot for your help! ☃️

**Comment by potiuk** [comment]
Maybe because of execution_delta ? As I understand it, tou need to run the task at the same schedule if you want to use execution_delta (and delta generally should be multiple of schedule interval (so in your case where you run daily task at 1 am, your sensor should also run at 1 am and your execution delta must be multiple of days). Otherwise sesis never going to be the same executon_date. Alternatively you can use `execution_date_fn` to set the execution date to check for.

BTW. I recommend you to use [data-awer scheduling ](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/datasets.html) if you want to have some dependencies between different dags. This is much easier to reason about and likely this is what you want to achieve.
